### PR TITLE
Fix when multiple PacBio files are given

### PR DESCRIPTION
**Pull request by muffato**
The bug occurred when there were multiple PacBio files. SAMTOOLS_FILTER was getting unsynchronised data from `ch_reads` and `PACBIO_FILTER.out.list`. The fix is to pair elements in those two channels before feeding SAMTOOLS_FILTER.

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/sanger-tol/readmapping/tree/master/.github/CONTRIBUTING.md)
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,12 @@
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [[1.2.2](https://github.com/sanger-tol/readmapping/releases/tag/1.2.2)] - [2024-05-23]
```
Comment (gq1) [inline CHANGELOG.md:6]: According to the instruction we give, should we add patch word in the release name?
https://pipelines.tol.sanger.ac.uk/docs/contributing/pipeline-release

**Comment by gq1** [comment]
Looks good for me.
But we need to fix the same problems in variantcalling pipeline.
https://github.com/sanger-tol/variantcalling/issues/77

**Review by gq1 — APPROVED** [review]

**Comment by gq1** [comment]
Merging to the main branch should be blocked by CI but not actually.
We should update the master branch to main in this CI: https://github.com/sanger-tol/readmapping/blob/a27e9ce841d230f3c1ca5086019a134a116bee18/.github/workflows/branch.yml#L6

**Comment by muffato** [comment]
> Merging to the main branch should be blocked by CI but not actually. We should update the master branch to main in this CI:
> 
> https://github.com/sanger-tol/readmapping/blob/a27e9ce841d230f3c1ca5086019a134a116bee18/.github/workflows/branch.yml#L6

Interesting find. I can see that the expected way is to name the branch `patch`.

I added a commit to this PR to fix master/main in `.github/workflows/branch.yml`

File: CHANGELOG.md (hunk 1 above)
Comment (muffato) [reply CHANGELOG.md:6]: Oh yes, absolutely

**Pull request merged**